### ISSUE TITLE
[BUG]: Test Drive Unlimited - Software render glitches when displaying highways.

### ISSUE THREAD

**Issue by GitHubProUser67** [PCSX2/pcsx2#11643]
### Describe the Bug

In the game Test Drive Unlimited, highways seems to suffer of a software render glitch that is not present when playing with any of the provided hardware renderers.

### Reproduction Steps

Load the provided savestate to observe the issue while driving.

If not using a savestate, simply drive under a highway.

### Expected Behavior

The highway should not have a weird line at the bottom of it, and flash colors while driving.

### PCSX2 Revision

v2.1.38

### Operating System

Windows 11

### CPU

I9-9900K

### GPU

RTX 3070

### GS Window Screenshots

![image](https://github.com/user-attachments/assets/1023f474-08d5-4a35-8b6f-9e1e7e2f97a9)

### Logs & Dumps

GS-Dump while observing the highway:

[Test Drive Unlimited_SLES-54466_20240729211722.zip](https://github.com/user-attachments/files/16418117/Test.Drive.Unlimited_SLES-54466_20240729211722.zip)

**Comment by F0bes** [comment]
Are you able to create a GS dump in roughly the same location but in hardware mode (so the issue isn't there) and upload it here?

**Comment by GitHubProUser67** [comment]
Yes of course, I did the next best thing, and took 2 GSdumps at the same place, one with software, the other with hardware : 

[TDU Gsdumps.zip](https://github.com/user-attachments/files/16573222/TDU.Gsdumps.zip)

**Comment by terremoth** [comment]
I managed to find the exact same place and position you are, and mine looks like this:

![WhatsApp Image 2024-08-22 at 02 04 27](https://github.com/user-attachments/assets/33b351c7-52c6-48ab-8912-2fe465dfece9)

**Comment by terremoth** [comment]
Here it is a single frame gs dump if it helps: 
[Test Drive Unlimited_SLUS-21490_20240822020642.gs.zip](https://github.com/user-attachments/files/16705652/Test.Drive.Unlimited_SLUS-21490_20240822020642.gs.zip)

and a save state file, if can help too:
[texture test drive save state.zip](https://github.com/user-attachments/files/16705638/texture.test.drive.save.state.zip)

Mine is SLUS-21490, Idk if this interfere some way or a lot.

**Comment by mirh** [comment]
Maybe #13240 could help

**Comment by refractionpcsx2** [comment]
This was fixed in #13240 mirh guessed correctly
